File: 백준/Silver/3273. 두 수의 합/두 수의 합.py
n = int(input())

arr = list(map(int, input().split()))

x = int(input())

arr.sort()

left = 0
right = n - 1

cnt = 0

while left < right:
    num = arr[left] + arr[right]

    if num == x:
        cnt += 1
        left += 1

    elif num > x:
        right -= 1
    else:
        left += 1

print(cnt)

File: 백준/Silver/3273. 두 수의 합/두 수의 합.js
const readline = require("readline");

const rl = readline.createInterface({
  input: process.stdin,
  output: process.stdout,
});

let n;
let arr;
let x;
rl.on("line", (line) => {
  if (!n) {
    n = parseInt(line);
  } else if (!arr) {
    arr = line.split(" ").map(Number);
  } else {
    x = parseInt(line);
  }
}).on("close", () => {
  arr.sort((a, b) => a - b);
  let left = 0;
  let right = n - 1;
  let cnt = 0;
  while (left < right) {
    const num = arr[left] + arr[right];
    if (num < x) {
      left++;
    } else if (num > x) {
      right--;
    } else if (num === x) {
      cnt++;
      left++;
    }
  }
  console.log(cnt);
});
